frontend/src/components/AddSpot.tsx

```diff
@@ -0,0 +1,227 @@
+import {
+    Box,
+    Button,
+    FormControl,
+    FormControlLabel,
+    FormLabel, List, ListItem,
+    Radio,
+    RadioGroup,
+    SelectChangeEvent,
+    Slider,
+    TextField,
+    Typography
+} from "@mui/material";
+
+import {
+    beachtypes,
+    months,
+    disciplines,
+    experiencesLevel,
+    hazards,
+    wavetypes,
+    windDirections, waterTemperatures
+} from "../statics/Statics";
+
+import {ChangeEvent, useEffect, useState} from "react";
+import CustomSelect from "./ui/CustomSelect";
+import {Spot} from "../models/Spot";
+import {Position} from "../models/Position";
+
+const emptySpot: Spot = {
+    id: "",
+    name: "empty",
+    disciplines: [],
+    waveTypes: [],
+    beachTypes: [],
+    experiencesLevel: [],
+    hazards: [],
+    bestMonths: [],
+    bestDirections: [],
+    waterTemperature: [],
+    parkingSpace: 0,
+    location: {lat: 0, lng: 0},
+    restrooms:"no"
+}
+
+const parkingSliderMarks = [
+    {
+        value: 0,
+        label: "few"
+    },
+    {
+        value: 1,
+        label: "enough"
+    },
+    {
+        value: 2,
+        label: "alot"
+    }
+]
+
+type AddSpotProps ={
+
+    pickedLocation:Position
+    handleCancel():void
+    handleSave(newSpot: Spot):void
+}
+
+export default function AddSpot(props:AddSpotProps) {
+    const [newSpot, setNewSpot] = useState<Spot>(emptySpot);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    useEffect(()=>{
+        setNewSpot({...newSpot, location:props.pickedLocation})
+    })
+
+    function handleInputChange(event: SelectChangeEvent<string[]>
+        | ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+        | Event) {
+        if (event.target) {
+            // There is actually an event.target no matter what event gets fired, there for we ignore
+            // This comes from the mui Slider [Event]
```

Comment:
What comes from where?
If "it" (the event?) comes from the Slider, why are there multiple EventOptions?